airbyte-ci/connectors/pipelines/pipelines/helpers/execution/run_steps.py

```diff
@@ -208,7 +208,10 @@ def _filter_skipped_steps(steps_to_evaluate: STEP_TREE, skip_steps: List[str], r
 
         else:
             steps_to_run.append(step_to_eval)
-
+    if not steps_to_run:
+        raise InvalidStepConfiguration(
```

Comment:
nit: I'm not sure we shall raise an exception here.
`steps_to_run` are currently static: they apply to all connectors selected by `airbyte-ci connectors`.

In https://github.com/airbytehq/airbyte/pull/38281 I'm slightly changing this logic, some connectors might run different steps according to the state of their metadata file.

It might conceptually mean that it's fine to have empty `steps_to_run`.
In any case that's not too problematic to raise ATM as step trees still always contains static steps to run by default (e.g. build, qa checks etc.)

Reply:
Hm. Actually, after trying to fix a unit test, I realized that the recursive looping makes this more complicated. We could check that the results_dict actually has step results in it after everything has executed, but there are many places where we'd have to update, and then decide if its okay if steps are skipped in different places (e.g. maybe migrate-to-poetry is a no-op if all steps are skipped for some valid reason, while i'd always expect test to have steps)

Since you're actively playing with this logic, i'll just remove it